akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -316,6 +317,50 @@ class Concat[T] private (override val shape: UniformFanInShape[T, T],
   override def named(name: String): Concat[T] = withAttributes(Attributes.name(name))
 }
 
+object Split {
+  /** Creates an Split using the given function */
+  def apply[In, A, B](f: In ⇒ Either[A, B]): Split[In, A, B] = {
+    val shape = new FanOutShape2[In, A, B]("Split")
+    new Split(shape, new FlexiRouteModule(
+      shape,
+      (s: FanOutShape2[In, A, B]) ⇒ new RouteLogic[In] {
+        override def initialState: State[_] = State[Any](DemandFromAll(s.out0, s.out1)) {
```

Comment:
Hmm, this is a limitation of the current API. This strictly speaking does not necessarily needs demand from all ports, after the predicate has been evaluated for an element.